tests/vmi_configuration_test.go

```diff
@@ -2599,30 +2599,31 @@ var _ = Describe("Configurations", func() {
 		)
 	})
 
-	Context("Custom PCI Adresses with slots and function", func() {
+	Context("Custom PCI Addresses configuration", func() {
+		// The aim of the test is to validate the configurability of a range of PCI slots
+		// on the root PCI bus 0. We would like to test slots 2..1a (slots 0,1 and beyond 1a are reserved).
+		// In addition , we test usage of PCI functions on a single slot
+		// by occupying all the functions 1..7 on random port 2.
+
+		addrPrefix := "0000:00" // PCI bus 0
+		numOfSlotsToTest := 24  // slots 2..1a
+		numOfFuncsToTest := 8
+		var vmi *v1.VirtualMachineInstance
 
-		It("should configure custom pci address across all slots", func() {
-			pciSlotsAddr := make([]string, 25)
-			for i := range pciSlotsAddr {
-				pciSlotsAddr[i] = fmt.Sprintf("%x", i+2)
-				if len(pciSlotsAddr[i]) == 1 {
-					pciSlotsAddr[i] = "0" + pciSlotsAddr[i]
-				}
-				pciSlotsAddr[i] = fmt.Sprintf("0000:00:%v.0", pciSlotsAddr[i])
-			}
-			vmi1 := tests.NewRandomFedoraVMIWitGuestAgent()
-			for i, pci := range pciSlotsAddr[:len(pciSlotsAddr)/2] {
-				vmi1.Spec.Domain.Devices.Disks = append(vmi1.Spec.Domain.Devices.Disks,
+		setupDisks := func(numOfAdditionalDisks, startIndex int, funcEnumerate bool, vmi *v1.VirtualMachineInstance) {
```

Comment:
One suggestion. `setupDisks` looks more complicated than it would have to. How about moving adding the disks out of this function and only assign the addresses in it? Also could you rename `funcEnumerate` into something like `assignToFunction`? Maybe even split this part and have two functions `func assignDisksToSlot` and `func assignDisksToFunctions`?

Reply:
@rmohr Done. Thank you for the suggestion!